mongo/mongo.go

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright 2022 SkyAPM org
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package mongo
+
+import (
+	"context"
+
+	"github.com/SkyAPM/go2sky"
+	"go.mongodb.org/mongo-driver/event"
+	agentv3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	// ComponentMongo ComponentID.
+	ComponentMongo int32 = 42
+
+	// ComponentMongoDB db.type.
+	ComponentMongoDB string = "MongoDB"
+
+	// Peer peer.
+	Peer string = "mongo:27017"
+)
+
+// Option custom option.
+type Option func(span go2sky.Span, evt *event.CommandStartedEvent)
+
+// Middleware mongo monitor.
+func Middleware(tracer *go2sky.Tracer, opts ...Option) *event.CommandMonitor {
+	spanMap := make(map[int64]go2sky.Span)
+	apmMonitor := &event.CommandMonitor{
+		Started: func(ctx context.Context, evt *event.CommandStartedEvent) {
+			span, _, err := tracer.CreateLocalSpan(ctx,
+				go2sky.WithSpanType(go2sky.SpanTypeEntry),
+				go2sky.WithOperationName(GetOpName(evt.CommandName)),
+			)
+			if err != nil {
+				return
+			}
+			span.SetPeer(Peer)
+			span.SetComponent(ComponentMongo)
+			span.SetSpanLayer(agentv3.SpanLayer_Database)
+			span.Tag(go2sky.TagDBType, ComponentMongoDB)
```

Comment:
You removed command tag?

Reply:
command tag is unpredictable. If necessary, can use Option to customize.

Reply:
Yes, I think an option is good.

Reply:
Should we wait for this new option?

Reply:
I don't have a preference. SkyWalking community just recommends the command tag.
So, I want to check with you, do you plan to add it?

Reply:
Ok, add command tag.